src/main/java/com/uber/cadence/samples/hello/TimerReschedule.java

```diff
@@ -0,0 +1,89 @@
+/*
+ *  Copyright 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Modifications copyright (C) 2017 Uber Technologies, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not
+ *  use this file except in compliance with the License. A copy of the License is
+ *  located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed on
+ *  an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package com.uber.cadence.samples.hello;
+
+import static com.uber.cadence.samples.common.SampleConstants.DOMAIN;
+
+import com.uber.cadence.client.WorkflowClient;
+import com.uber.cadence.client.WorkflowClientOptions;
+import com.uber.cadence.serviceclient.ClientOptions;
+import com.uber.cadence.serviceclient.WorkflowServiceTChannel;
+import com.uber.cadence.worker.Worker;
+import com.uber.cadence.worker.WorkerFactory;
+import com.uber.cadence.workflow.SignalMethod;
+import com.uber.cadence.workflow.Workflow;
+import com.uber.cadence.workflow.WorkflowMethod;
+
+public class TimerReschedule {
+
+  static final String TASK_LIST = "TimerReschedule";
+
+  public interface TimerRescheduleWorkflow {
+    @WorkflowMethod(executionStartToCloseTimeoutSeconds = 10, taskList = TASK_LIST)
+    String start();
+
+    @SignalMethod
+    void extend();
+  }
+
+  public static class TimerRescheduleWorkflowImpl implements TimerRescheduleWorkflow {
+
+    private long expirationTimeMs;
+    private static long TIMEOUT = 60 * 1000; // 1 minute
+
+    @Override
+    public String start() {
+      expirationTimeMs = Workflow.currentTimeMillis() + TIMEOUT;
```

Comment:
may be move this to constructor to mimic exactly.

Reply:
OR you can simply copy paste the timer from the https://sourcegraph.indeed.tech/code.corp.indeed.com/workflow/wayfinder-expedition-host/-/blob/server/src/main/java/com/indeed/wayfinder/expedition/host/time/ExpirationTimer.java?L24 to maintain accuracy.